File: Examples/Node.js/Document/SplitPresentation.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var SlidesApi = require('asposeslidescloud');
var StorageApi = require('asposestoragecloud');
var configProps = require('../Data/config.json');
var data_path = '../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var outFolder = configProps.out_folder;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose.Slides API SDK
var slidesApi = new SlidesApi(config);

// Set input file name
var name = "sample-input.pptx";
var from = 2;
var to = 3;
var format = "png";

var versionId = null;
var storage = null;
var folder = null;

try {
// Upload source file to aspose cloud storage
storageApi.PutCreate(name, versionId, storage, data_path + name , function(responseMessage) {

	assert.equal(responseMessage.status, 'OK');

	// Invoke Aspose.Slides Cloud SDK API to split all or specific slides of a presentation file to specify format
	slidesApi.PostSlidesSplit(name, null, null, to, from, null, format, storage, folder, function(responseMessage) {
			assert.equal(responseMessage.status, 'OK');

			responseMessage.body.SplitResult.Slides.forEach(function(splitDoc) {
				var urlString = splitDoc.Href;
				var splittedFileName = urlString.split('/').pop();
				
				// Download file from storage server
				storageApi.GetDownload(splittedFileName, null, null, function(responseMessage) {
					assert.equal(responseMessage.status, 'OK');
					var writeStream = fs.createWriteStream(data_path + splittedFileName);
					writeStream.write(responseMessage.body);
					});
				});

			});
	});

}catch (e) {
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1

File: Examples/Perl/Document/SplitPresentation.pl
#ExStart:1
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeStorageCloud::StorageApi;
use AsposeStorageCloud::ApiClient;
use AsposeStorageCloud::Configuration;

use AsposeSlidesCloud::SlidesApi;
use AsposeSlidesCloud::ApiClient;
use AsposeSlidesCloud::Configuration;

my $configFile = '../Config/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);
my $data_path = '../../../Data/';
my $out_path = $configProps->{'out_folder'};

$AsposeSlidesCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeSlidesCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposeSlidesCloud::Configuration::debug = 1;

$AsposeStorageCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeStorageCloud::Configuration::api_key = $configProps->{'api_key'};

# Instantiate Aspose.Storage and Aspose.Slides API SDK
my $storageApi = AsposeStorageCloud::StorageApi->new();
my $slidesApi = AsposeSlidesCloud::SlidesApi->new();

# Set input file name
my $name = 'sample-input.pptx';
my $from = 2;
my $to = 3;
my $format = 'png';
	
# Upload file to aspose cloud storage 
my $response = $storageApi->PutCreate(Path => $name, file => $data_path.$name);

# Invoke Aspose.Slides Cloud SDK API to split all or specific slides of a presentation file to specify format         
$response = $slidesApi->PostSlidesSplit(name => $name, from=>$from, to=>$to, format=>$format);

if($response->{'Status'} eq 'OK'){
	foreach my $splitDoc (@{$response->{'SplitResult'}->{'Slides'}}){
		my $urlString = $splitDoc->{'Href'};
		my ($file) = $urlString =~ m|^.*/(.*)$|;
		print "\n File :: $file";
		my $output_file = $out_path. $file;
		$response = $storageApi->GetDownload(Path => $file);;
		write_file($output_file, { binmode => ":raw" }, $response->{'Content'});
	}	
}
#ExEnd:1